py/server/deephaven/experimental/partitioned_table_service.py

```diff
@@ -0,0 +1,326 @@
+        """ Provides the new partitions for the table with the given table key to the table service in the engine.
+
+        Args:
+            table_key (TableKey): the table key
+            callback (jpy.JType): the Java callback function with two arguments: a table location key of the new
+                partition and an array of byte buffers that contain the arrow schema and the serialized record batches
+                for the partition column values
+        """
+        def callback_proxy(pt_location_key, pt_table):
+            j_tbl_location_key = _JTableLocationKeyImpl(pt_location_key)
+            if pt_table is None:
+                callback.apply(j_tbl_location_key, jpy.array("java.nio.ByteBuffer", []))
+            else:
+                if pt_table.num_rows != 1:
+                    raise ValueError("The number of rows in the pyarrow table for partition column values must be 1")
+                bb_list = [jpy.byte_buffer(rb.serialize()) for rb in pt_table.to_batches()]
+                bb_list.insert(0, jpy.byte_buffer(pt_table.schema.serialize()))
+                callback.accept(j_tbl_location_key, jpy.array("java.nio.ByteBuffer", bb_list))
+
+        return self._backend.subscribe_to_new_partitions(table_key, callback_proxy)
+
+    def _partition_size(self, table_key: TableKey, table_location_key: PartitionedTableLocationKey, callback: jpy.JType):
+        """ Provides the size of the partition with the given table key and partition location key to the table service
+        in the engine.
+
+        Args:
+            table_key (TableKey): the table key
+            table_location_key (PartitionedTableLocationKey): the partition location key
+            callback (jpy.JType): the Java callback function with one argument: the size of the partition in number of
+                rows
+        """
+        def callback_proxy(size):
+            callback.accept(size)
+
+        self._backend.partition_size(table_key, table_location_key, callback_proxy)
+
+    def _subscribe_to_partition_size_changes(self, table_key: TableKey, table_location_key: PartitionedTableLocationKey,
+                                             callback: jpy.JType) -> jpy.JType:
+        """ Provides the changed size of the partition with the given table key and partition location key to the table
+        service in the engine.
+
+        Args:
+            table_key (TableKey): the table key
+            table_location_key (PartitionedTableLocationKey): the partition location key
+            callback (jpy.JType): the Java callback function with one argument: the size of the partition in number of
+                rows
+        """
+        def callback_proxy(size):
+            callback.accept(size)
+
+        return self._backend.subscribe_to_partition_size_changes(table_key, table_location_key, callback_proxy)
+
+    def _column_values(self, table_key: TableKey, table_location_key: PartitionedTableLocationKey, col: str, offset: int,
```

Comment:
This is the only occurrence of this function, and it is private.

Reply:
It is private because it is only called by the Java backend and should not be visible to the Python users.

Reply:
Then there should be a docstring making this clear.

Reply:
If there are any other similar cases, the docstring or comment should clarify that they are used by java.